pkg/agent/controller/networkpolicy/cache.go

```diff
@@ -143,7 +140,7 @@ type ruleCache struct {
 
 	policyMapLock sync.RWMutex
 	// policyMap is a map using NetworkPolicy UID as the key.
-	policyMap map[string]*v1beta.NetworkPolicyReference
+	policyMap map[string]*v1beta.NetworkPolicy
```

Comment:
Will this cause additional storage cost? Data may be redundant between ruleCache.policyMap and ruleCache.rules?

Reply:
Thanks for the reminder. You are right. We didn't store the policy directly to avoid redundant storage at the very beginning because we didn't need it. However, more and more features require it now. It becomes inconvenient to construct the policy based on rules and storing each field of policy to rule is redundant too. I added a TODO to reduce the storage redundancy, we could stop storing the policy's fields in rules, and use a pointer to the policy instead.